khal/icalendar.py

```diff
@@ -72,8 +72,18 @@ def split_ics(ics: str, random_uid: bool=False, default_timezone=None):
             events_grouped[item['UID']].append(item)
         else:
             continue
-    return [ics_from_list(events, tzs, random_uid, default_timezone) for uid, events in
-            sorted(events_grouped.items())]
+    out = []
+    saved_exception = None
+    for uid, events in sorted(events_grouped.items()):
+        try:
+            ics = ics_from_list(events, tzs, random_uid, default_timezone)
+        except Exception as exception:
+            logger.warn(f'Error when trying to import the event {uid}')
+            saved_exception = exception
+        out.append(ics)
+    if saved_exception:
+        raise saved_exception
```

Comment:
This will only raise the last exception (which I guess is fine?).

Maybe we can use an [`ExceptionGroup`](https://docs.python.org/3/library/exceptions.html#exception-groups) in a few years when Python 3.11 has trickled down to most distributions.

Reply:
Yeah, the last is probably enough. If we wanted more detail, we could include `{exception}` or `{exception!r}` in the warning?